Modules/DevTools/TestFramework/TestLibraries/Any/src/Any.Codeunit.al

```diff
@@ -220,9 +221,20 @@ codeunit 130500 "Any"
     procedure SetSeed(NewSeed: Integer)
     begin
         Seed := NewSeed;
+        SeedSet := true;
         Randomize(Seed);
     end;
 
+    /// <summary>.
+    /// Sets the default Seed for Pseudo-random number generation (no. of milliseconds since midnight of today).
+    /// Setting this value will change the numbers returned.
+    /// </summary>
+    procedure SetDefaultSeed()
```

Comment:
This method should not be added. Main reason is - it uses a random seed, which means that it will not be possible to reproduce the failure that occurred in the gate. This is rather important, for this reason we always set a seed to a specific number so the numbers are pseudo-random.

We should not encourage the usage of the randomly generated numbers, since it will cause issues with debugging down the road.

If anyone wants to use a random generator they can generate the seed randomly and assign it via SetSeed. You can also run a test in the  for loop and increment the seed if you want to test the different seeds and flush out any rounding issues and so on...

Reply:
> Main reason is - it uses a random seed, which means that it will not be possible to reproduce the failure that occurred in the gate.

I agree with you about the debuggability, you are absolutely right. But there are scenarios where a fixed seed is cumbersome. I'm thinking about an extensive test library that has various functions that generates random data. Those functions have a local copy of `Any`. And therefore every time I call it, I get the same number because I always start at the same seed. So I really only have three options:
- I pass around a "root" instance of `Any` as a parameter to the function, which is obviously not very handy
- I have a global `Any` in my library codeunit, but that means that I always must make sure that it is correctly initialized before I call any function
- I use `SetSeed` to initialize my local `Any` instance. The way that `Randomize` does initialize the seed is exactly what I would do in that case. It would therefore be handy, if we had access to that same behavior in `Any`. Otherwise I need to reimplement the same logic in all my test libraries.

> We should not encourage the usage of the randomly generated number

I agree on this as well. But adding a method does not encourage anything IMHO. In most cases the developer will not even call `SetSeed` or any other method of initializing it, because seeding it with 1 is fine. If, on the other hand, I don't want that, I should be able to reuse existing logic.

> If anyone wants to use a random generator they can generate the seed randomly and assign it via SetSeed.

Same as above. That is exactly my point. `Randomize` already does this for us. But the codeunit `Any` abstracts this away. I agree that we can do this on or own, but it's already there, it already does exactly what we need.